fix: improve type safety in type definitions

Myestery opened the pull request:
## Summary

- Replace `any` types with proper TypeScript types in core type definitions
- Add generics to `SettingParams`, `setting.get<T>()`, and `setting.set<T>()` for type-safe setting access
- Add `NodeExecutionOutput` interface for `onExecuted` callback
- Change function parameters from `any` to `unknown` where appropriate

## Type Research

Performed GitHub code search across custom node repositories to understand actual usage patterns:

**`onExecuted` output properties** (used in rgthree-comfy, ComfyUI-KJNodes, ComfyUI-ExLlama-Nodes, comfy_mtb, etc.):
- `output.text` - string or string array for text display nodes
- `output.images`, `output.audio`, `output.video` - media outputs
- `output.ui.items` - complex debug/preview data with `input`, `text`, `b64_images`

**`extensionManager.setting.get/set`** (used in ComfyUI-Crystools, ComfyUI-Copilot, etc.):
- Returns various types (boolean, number, string, objects)
- Now uses generics: `setting.get<boolean>('MyExt.Setting')`

**`ComfyExtension` custom properties** (used in rgthree-comfy, ComfyUI-Manager):
- `aboutPageBadges`, `commands`, custom methods
- Kept as `any` index signature since extensions add arbitrary properties

## Changes

| File | Change |
|------|--------|
| `extensionTypes.ts` | Generic `setting.get<T>()` and `setting.set<T>()`, typed Toast options |
| `litegraph-augmentation.d.ts` | `onExecuted(output: NodeExecutionOutput)` |
| `metadataTypes.ts` | GLTF index signatures `any` → `unknown` |
| `apiSchema.ts` | New `NodeExecutionOutput` interface |
| `settings/types.ts` | `SettingOnChange<T>`, `SettingMigration<T>`, `SettingParams<TValue>` |
| `nodeDefSchema.ts` | `validateComfyNodeDef(data: unknown)` |
| `workflowSchema.ts` | `isSubgraphDefinition(obj: unknown)` |
| `telemetry/types.ts` | `checkForCompletedTopup(events: AuditLog[])` |

## Test plan

- [x] `pnpm typecheck` passes
- [x] `pnpm test:unit` passes (3732 tests)
- [x] `pnpm lint` passes

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7337-fix-improve-type-safety-in-type-definitions-2c66d73d365081bdbc30e916cac607d6) by [Unito](https://www.unito.io)

Myestery committed:
fix: improve type safety in core type definitions

- extensionTypes.ts: Add generics to setting.get<T>() and setting.set<T>()
  for type-safe setting access. Change ToastMessageOptions to use proper
  Vue class binding types instead of any.

- litegraph-augmentation.d.ts: Type onExecuted callback with
  NodeExecutionOutput interface instead of any. Based on code search of
  custom nodes (rgthree-comfy, ComfyUI-KJNodes, comfy_mtb, etc.) which
  access output.text, output.images, output.ui.items properties.

- metadataTypes.ts: Change GLTF index signatures from any to unknown.

Myestery committed:
fix: add NodeExecutionOutput type for onExecuted callback

Add explicit interface for node execution output based on code search
of custom node extensions:
- audio/images/video/animated: Core media properties from zOutputs
- text: Used by ShowText, Debug nodes (ComfyUI-KJNodes, ExLlama-Nodes)
- ui.items: Used by debug nodes (comfy_mtb) for complex output

This avoids using z.infer<typeof zOutputs> which has strict index
signature from passthrough() that causes type errors.

Myestery committed:
fix: add generics to SettingParams for type-safe callbacks

- Add SettingOnChange<T> and SettingMigration<T> generic types
- SettingParams<TValue> now uses generics for defaultValue, onChange,
  and migrateDeprecatedValue
- Update settingStore to handle callable defaultValue with proper cast
- Update coreSettings migrateDeprecatedValue to use typed parameters
- Change SettingCustomRenderer and FormItem.attrs to use unknown

Myestery committed:
fix: improve type safety in Zod schemas

- nodeDefSchema: Change validateComfyNodeDef parameter from any to unknown
- workflowSchema: Change isSubgraphDefinition parameter from any to unknown
  with proper type narrowing

Myestery committed:
fix: type checkForCompletedTopup with AuditLog[]

Change events parameter from any[] to AuditLog[] to match the
actual implementation in topupTracker.ts.

Myestery committed:
fix: update call sites to use new typed APIs

- useToolManager: Use setting.get<T>() generic instead of type assertion
- previewAny: Handle message.text as string | string[] | undefined

Myestery committed:
Merge branch 'main' into fix/type-safety-pr1-definitions

DrJKL reviewed (commented):
Thanks for working on this!

A few small comments, but every `any` eliminated is a mitzvah.

Myestery committed:
fix: use z.infer for NodeExecutionOutput instead of manual interface

Simplify by using Zod's inferred type directly. The passthrough()
index signature works correctly for the onExecuted callback use case.

Myestery committed:
Merge branch 'main' into fix/type-safety-pr1-definitions

Myestery committed:
fix: improve type safety for message handling and extend output schema with optional text field

DrJKL commented on src/platform/settings/types.ts:
```diff
@@ -16,30 +16,29 @@ type SettingInputType =
 }
 
-export interface SettingParams<TValue = unknown> extends FormItem {
+export interface SettingParams<TValue = any> extends FormItem {
```
@Myestery WDYT?
Or is this something you'll be doing in the next PR?

Myestery replied:
yeah, as we progress the types will become clearer
